### ISSUE TITLE
unable to see kubeml python logging

### ISSUE THREAD

**Issue by nwangfw** [DiegoStock12/kubeml#4]
Hi @DiegoStock12,

I am trying to modify the network class in the kubeml python library (i.e., saving the optimizer) but it didn't work somehow.  It would be great if I can read the python logs but I couldn't print logs out.

The following is what I did,

(1) I modified the dockerfile in the environment folder and set the gunicorn logging level to debug with the following command,
```
CMD ["gunicorn", "--log-level", "debug", "-b", "0.0.0.0:8888", "-t", "2000", "-w 1", "server:app"]
```
(2) I then logged into one pod to check the training logs with the following command,
```
 kubectl -n fission-function logs poolmgr-torch-default-12086650-56b584f4fc-8j5kx --container torch
```
I suppose that the python logging should also be printed here. However, I could only get the gunicorn logs,

```
[2023-01-11 15:37:17 +0000] [1] [DEBUG] Current configuration:
  config: ./gunicorn.conf.py
  wsgi_app: None
  bind: ['0.0.0.0:8888']
  backlog: 2048
  workers: 1
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 2000
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /app
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 0
  group: 0
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: None
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: -
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  syslog_addr: udp://localhost:514
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: server:app
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x7f39077539d8>
  on_reload: <function OnReload.on_reload at 0x7f3907753ae8>
  when_ready: <function WhenReady.when_ready at 0x7f3907753bf8>
  pre_fork: <function Prefork.pre_fork at 0x7f3907753d08>
  post_fork: <function Postfork.post_fork at 0x7f3907753e18>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x7f3907753f28>
  worker_int: <function WorkerInt.worker_int at 0x7f39077720d0>
  worker_abort: <function WorkerAbort.worker_abort at 0x7f39077721e0>
  pre_exec: <function PreExec.pre_exec at 0x7f39077722f0>
  pre_request: <function PreRequest.pre_request at 0x7f3907772400>
  post_request: <function PostRequest.post_request at 0x7f3907772488>
  child_exit: <function ChildExit.child_exit at 0x7f3907772598>
  worker_exit: <function WorkerExit.worker_exit at 0x7f39077726a8>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x7f39077727b8>
  on_exit: <function OnExit.on_exit at 0x7f39077728c8>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2023-01-11 15:37:17 +0000] [1] [INFO] Starting gunicorn 20.1.0
[2023-01-11 15:37:17 +0000] [1] [DEBUG] Arbiter booted
[2023-01-11 15:37:17 +0000] [1] [INFO] Listening at: http://0.0.0.0:8888 (1)
[2023-01-11 15:37:17 +0000] [1] [INFO] Using worker: sync
[2023-01-11 15:37:17 +0000] [9] [INFO] Booting worker with pid: 9
[2023-01-11 15:37:17 +0000] [1] [DEBUG] 1 workers

```
Can you let me know which part of my command is wrong and how to get the kubeml python logging printed? Many thanks for your help!

**Comment by DiegoStock12** [comment]
Hi @nwangfw! 

This should work without any problems and you should see logs both from fission, the kubeml python library and the customer logs that you decide to set in your code. 

I am not sure of the configurations you're using regarding parallelism of the function and the one you're configuring in the kubeml job. To make sure that you can get all the logs no matter in which container of the pool they're being dumped, you might want to try debugging with the `fission function` cli command, [here](https://fission.io/docs/reference/fission-cli/fission_function_log/) you can find the reference.

There is an option (`--all-pods`) to join the logs of all containers of a function into the same terminal output, which can be very useful.

Let me know if you're able to debug using this.

Regards,
Diego

**Comment by nwangfw** [comment]
Hi Diego,

Thanks so much for your help and suggestions. I figured it out with fission cli.

I previously simply assume that a new job is assigned to the newly created pods and I always log into those pods.  However, fission doesn't work in that way and the newly creately pods are general pods to ensure the warm up pool has a certain size.

After reading some fission instruction, I found that I can use the following command to locate the pods assigned for the task,
```
fission function pod --name=<function name>
```

Thanks again for your valuable time and this interesting project. Will dive deeper into it.
